interfaces/builtin/gpio_memory_control.go

```diff
@@ -0,0 +1,51 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const gpioMemoryControlSummary = `allows write access to all gpio memory`
+
+const gpioMemoryControlBaseDeclarationSlots = `
+  gpio-memory-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const gpioMemoryControlConnectedPlugAppArmor = `
+# Description: Allow writing to /dev/gpiomem on kernels that provide it.
+#
+/dev/gpiomem rw,
+`
+
+var gpioMemoryControlConnectedPlugUDev = []string{`KERNEL=="gpiomem"`}
+
+func init() {
+	registerIface(&commonInterface{
+		name:                  "gpio-memory-control",
```

Comment:
I'm conflicted on the name. On the one hand /dev/gpiomem gives precisely access to all GPIO memory so `gpio-memory-control` makes a lot of sense. However, there are several GPIO kernel systems and I wonder if it makes sense to have this interface reflect that this is for one, /dev/gpiomem, and therefore the interface could be named `gpiomem-control`. I think I prefer what you have, `gpio-memory-control`, but please get signoff from @niemeyer on the name.

Reply:
I am similarly conflicted, and have no strong opinions.